### PR TITLE
Optimize reindexing for document checkout/checkin cycles

### PR DESCRIPTION
**Pull request by Rotonen**
* I've plugged the D'n'D upload reindexing hole
* Only reindex the `checked_out` index

### CODE REVIEW

File: opengever/document/checkout/manager.py [hunk 1]
```diff
@@ -89,7 +89,9 @@ def checkout(self):
         self.annotations[CHECKIN_CHECKOUT_ANNOTATIONS_KEY] = user_id
 
         # finally, reindex the object
-        self.context.reindexObject()
+        self.context.reindexObject(idxs=(
+            'checked_out',
+            ))
```
Comment (jone) [inline opengever/document/checkout/manager.py]: Should we update metadata here?

I'd prefer to have an explicit `update_metadata=True or False` here in order to indicate that we thought about it and decided for one way or the other.

Comment (Rotonen) [reply opengever/document/checkout/manager.py]: AFAICS the metadata edits are handled by a form so I'll just amend an explicit False here.

Comment (Rotonen) [reply opengever/document/checkout/manager.py]: @jone `object.reindexObject(idxs=[])` does not have a keyword argument for updating metadata - should I also port this to use `portal_catalog.reindexObject(obj, idxs=[], update_metadata=False`?

Comment (jone) [reply opengever/document/checkout/manager.py]: @Rotonen yes, makes sense 👍

Comment (Rotonen) [reply opengever/document/checkout/manager.py]: @jone actually we do need to update the metadata as per other tests failing when this is not done and this is thus apparently always implicitly done when one uses `object.reindexObject()`.

Comment (jone) [reply opengever/document/checkout/manager.py]: Ah, there is a `checked_out` metadata. I pitty that it is not possible to only update selected metadata..

**Comment by phgross** [comment]
@Rotonen still WIP or ready to merge ?

**Comment by Rotonen** [comment]
@jone did I miss anything here? I think it's better than it was, but I'm not sure if this actually closes #3124 .

**Comment by jone** [comment]
Looks good 👍 it solves what's described in #3124.
A changelog entry would be nice.

**Comment by Rotonen** [comment]
@jone ready for rereview.

**Comment by phgross** [comment]
@Rotonen please rebase ...

**Comment by Rotonen** [comment]
@phgross rebased.

**Pull request merged**